### PR TITLE
Improve Mailer Documentation

### PR DESCRIPTION
**Pull request by tubbo**
Direct readers to ActionMailer resources when they're looking to create
new mailers rather than style or modify existing ones. Also added some
information about unit testing mailer classes.

### CODE REVIEW

File: docs/source/articles/system-emails.html.md [hunk 1]
```diff
@@ -99,4 +99,38 @@ Rails provides functionality to [preview emails in a browser](http://api.rubyonr
...
+
+(If you generated a mailer with `rails generate mailer`, this file should already exist, so just change the contents of the existing file into what's described above.)
+
+You can now use assertions to determine whether mails have been sent:
+
+```ruby
+def test_mail_was_sent
+  assert_emails(1) do
+    valid_product = create_product
+    valid_product.send_an_email! # calls YourMailer.some_email.deliver_later
```
Comment (bencrouse) [inline docs/source/articles/system-emails.html.md]: A product sending an email seems like an odd choice here.

File: docs/source/articles/system-emails.html.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ excerpt: System emails are sent from both the admin and storefront engines. All
 
 # System Emails
 
-System emails are sent from both the admin and storefront engines. All system emails may be customized like any other view in the system, by first [overriding](overriding.html) the mailer views and associated stylesheets.
+System emails are sent from both the admin and storefront engines. All system emails may be customized like any other view in the system, by first [overriding](overriding.html) the mailer views and associated stylesheets. Emails sent from the system use Rails' [Action Mailer](https://guides.rubyonrails.org/action_mailer_basics.html#) to do all the heavy lifting, so if you're making a new mailer, be sure to read that guide for more information.
```
Comment (chriscressman) [inline docs/source/articles/system-emails.html.md]:
```suggestion
System emails are sent from both the admin and storefront engines. All system emails may be customized like any other view in the system, by first [overriding](overriding.html) the mailer views and associated stylesheets. Emails are sent from the system use Rails' [Action Mailer](https://guides.rubyonrails.org/action_mailer_basics.html#) to do all the heavy lifting, so if you're making a new mailer, be sure to read that guide for more information.
```

Comment (mttdffy) [reply docs/source/articles/system-emails.html.md]: minor, but proper noun possessive use `'s', so it should be Rails's

https://www.proofreadnow.com/blog/bid/84201/Possessives-Proper-Nouns

**Comment by chriscressman** [comment]
looks good, thanks @tubbo

**Pull request merged**